src/pubsub/src/subscriber/builder.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::stub::Stub;
+use std::sync::Arc;
+
+/// Builder for the `client::Subscriber::streaming_pull` method.
+pub struct StreamingPull<S>
+where
+    S: Stub,
+{
+    // TODO(#4061) - Use a dynamic stub to remove the generic.
+    pub(crate) inner: Arc<S>,
+    pub(crate) subscription: String,
+    pub(crate) ack_deadline_seconds: i32,
+    pub(crate) max_outstanding_messages: i64,
+    pub(crate) max_outstanding_bytes: i64,
+}
+
+impl<S> StreamingPull<S>
+where
+    S: Stub,
+{
+    pub(crate) fn new(inner: Arc<S>, subscription: String) -> Self {
+        Self {
+            inner,
+            subscription,
+            ack_deadline_seconds: 10,
+            max_outstanding_messages: 1000,
+            max_outstanding_bytes: 100 * 1024 * 1024,
+        }
+    }
+
+    /// Sets the ack deadline to use for the stream.
+    ///
+    /// This value represents how long the application has to ack or nack an
+    /// incoming message. Note that this value is independent of the deadline
+    /// configured on the server-side subscription.
+    ///
+    /// The minimum deadline you can specify is 10 seconds. The maximum deadline
+    /// you can specify is 600 seconds (10 minutes).
```

Comment:
What do you mean by "can" here? It seems like you can set to any values. Will something error? Will it get truncated?